### PR TITLE
Don't allow pressing the same button many times

### LINKED ISSUE #1949: It's possible to add multiple elements from one 'add element view'
Steps to reproduce
* Tap Add Element in the Page Editor
* Quickly tap the image icon as many times as you can before the Add Element view closes
* Check how many toucans you added, I'm able to add up to 4

You should only be able to add one element at a time.

**Update**
If you add two elements in this way, you will only ever be able to select one of them. You can move the one you can't select, but you can never delete or edit it.

### PR DESCRIPTION
**Pull request by errietta**
Closes #1949

### CODE REVIEW

File: www_src/pages/page/page.jsx [hunk 1]
```diff
@@ -164,6 +166,10 @@ var Page = React.createClass({
           state.currentElementId = id;
         }
         this.setState(state);
+
+        setTimeout(function() {
```
Comment (thisandagain) [inline www_src/pages/page/page.jsx]: Why the 1000ms timeout?

Comment (errietta) [inline www_src/pages/page/page.jsx]: I had an issue with the menu taking a few ms to hide, during which you could again click the buttons

Comment (thisandagain) [inline www_src/pages/page/page.jsx]: Gotcha. In that case, I'd recommend we match it to the transition time in the CSS (`0.2s`) and add a comment that explains why JIK that animation value ever gets changed.

**Comment by thisandagain** [comment]
R+!

File: www_src/pages/page/page.jsx [hunk 2]
```diff
@@ -164,6 +166,14 @@ var Page = React.createClass({
           state.currentElementId = id;
         }
         this.setState(state);
+
+        //Ensure we don't reach a race conditions where the buttons are
+        //re-enabled before the menu is hidden. 200ms matches the css animation
+        //speed.
+
+        setTimeout(function() {
```
Comment (Pomax) [inline www_src/pages/page/page.jsx]: this should not have been here. This should not be a timeout, but should have been triggered by the child element being done doing what it's doing, informing its parent that it's now done and the buttons can be shown again. In traditional HTML+JS you would use a timeout, but in React, using a timeout is a sign that lifecycles are not being respected.

Comment (thisandagain) [inline www_src/pages/page/page.jsx]: Can you please provide some guidance / links / resources for @errietta to take a look at?

Comment (Pomax) [inline www_src/pages/page/page.jsx]: the best place to start is to work through the "getting started" and "tutorial" sections on http://facebook.github.io/react, to get an understanding of how React wants you to structure things, and how to make use of the life cycle functions to trigger things. After that it depends a little on how much you picked up, but I'd be happy to help find further tutorials/resources on how to do specific things the "Reacty" way after running through React's own tutorial.

**Pull request merged**